### DISCUSSION TITLE
Why the last Concat has a huge impact on accuracy

### DISCUSSION THREAD

**Discussion by freedom-cui** [ultralytics/ultralytics#15762] [Q&A]
hello @glenn-jocher :
Thank you so much for all your contributions to the advancement of technology to you and your team!

Now there's a problem that bothers me a lot when I quantize the YOLO V8n model using the PTQ approach

When I just quantize the model from fp32 to fp16, there is not a big drop in accuracy

But when I use 8-bit quantization for all parameters, I can't detect any results at all

So I modified the YOLO V8n network structure by taking out the last concat：
The network structure before and after the modification is shown in the following figure.
![QQ_1724378030376](https://github.com/user-attachments/assets/10ca7c99-e2c3-46c6-8984-af0f77395ce7)

The modified network output is as follows
![068b0ca97936b3606924aa4063b54f22](https://github.com/user-attachments/assets/acc0df87-fe91-46d0-a4c3-a2ceaa904115)

So far the 8-bit quantization results are correct, and I'm wondering why the last concat has such a big impact on the 8-bit quantization.

Looking forward to your reply, thanks a lot!

**Comment by glenn-jocher** [comment]
@laogonggong847 thank you for your detailed explanation and kind words. The significant impact of the last Concat layer on 8-bit quantization accuracy is likely due to the sensitivity of concatenation operations to quantization errors. These operations can amplify small quantization errors, leading to a substantial drop in performance. You might want to explore mixed-precision quantization or layer-specific quantization strategies to mitigate this issue.

**Reply by freedom-cui** [reply]
Yes, can you give me more details on why this happens when quantizing mixed precision?

Thank you~

**Reply by glenn-jocher** [reply]
@laogonggong847 quantizing mixed precision can lead to accuracy drops because different layers may have varying sensitivity to quantization errors, and concatenation operations can amplify these errors. Adjusting the quantization strategy for specific layers might help mitigate this issue.
